tests/test_endpoint.py

```diff
@@ -0,0 +1,140 @@
+LSST Project (http://www.lsst.org/).
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the LSST License Statement and
+the GNU General Public License along with this program. If not,
+see <http://www.lsstcorp.org/LegalNotices/>.
+"""
+from __future__ import absolute_import, division, print_function
+import unittest
+
+import numpy as np
+import astshim
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+
+def makeRawArrData(nPoints, nAxes, delta=0.123):
+    return np.array([[i + j*delta for j in range(nAxes)] for i in range(nPoints)])
+
+def makeRawPointData(nAxes, delta=0.123):
+    return [i*delta for i in range(nAxes)]
+
+class EndpointTestCase(unittest.TestCase):
+
+    def testSpherePointEndpoint(self):
+        endpoint = afwGeom.SpherePointEndpoint()
+        self.checkEndpointBasics(endpoint=endpoint, pointType=afwGeom.SpherePoint, nAxes=2)
+
+        for doPermute in (False, True):
+            frame = astshim.SkyFrame()
+            if doPermute:
+                frame.permAxes([2, 1])
+                self.assertEqual(frame.getLonAxis(), 2)
+                self.assertEqual(frame.getLatAxis(), 1)
+            else:
+                self.assertEqual(frame.getLonAxis(), 1)
+                self.assertEqual(frame.getLatAxis(), 2)
+            endpoint.normalizeFrame(frame)
+            # the normalized frame always has axis in order Lon, Lat
+            self.assertEqual(frame.getLonAxis(), 1)
+            self.assertEqual(frame.getLatAxis(), 2)
+
+        badFrame = astshim.Frame(2)
+        with self.assertRaises(Exception):
+            endpoint.normalizeFrame(badFrame)
+
+        newFrame = endpoint.makeFrame()
+        self.assertEqual(type(newFrame), astshim.SkyFrame)
+        self.assertEqual(newFrame.getLonAxis(), 1)
+        self.assertEqual(newFrame.getLatAxis(), 2)
+
+    def testPoint2Endpoint(self):
+        endpoint = afwGeom.Point2Endpoint()
+        self.checkEndpointBasics(endpoint=endpoint, pointType=afwGeom.Point2D, nAxes=2)
+
+        frame1 = astshim.Frame(2)
+        endpoint.normalizeFrame(frame1)  # should not raise
+        badFrame = astshim.SkyFrame()
+        with self.assertRaises(Exception):
+            endpoint.normalizeFrame(badFrame)
+
+    def testPoint3Endpoint(self):
+        endpoint = afwGeom.Point3Endpoint()
+        self.checkEndpointBasics(endpoint=endpoint, pointType=afwGeom.Point3D, nAxes=3)
+
+        frame1 = astshim.Frame(2)
+        endpoint.normalizeFrame(frame1)  # should not raise
```

Comment:
OK, though `SpecificException` in this case seems out of place. It should pass and anything else is an error.